### PR TITLE
Feature/mutable_state

### PR DESCRIPTION
**Pull request by phischu**

### CODE REVIEW

**Comment by b-studios** [comment]
As soon as this branch is in a somewhat stable state, I would be happy to merge it. You can of course continue to work on it, but this way it is easier to deal with upstream changes (then you don't need to rebase so many commits every time master changes).

**Comment by serkm** [comment]
This is ready to merge now.

File: effekt/shared/src/main/scala/effekt/generator/chez/ChezScheme.scala [hunk 1]
```diff
@@ -132,7 +132,7 @@ trait ChezScheme {
       }
       chez.Handle(handlers, toChez(body))
 
-    case Region(body) => chez.Builtin("with-region", toChez(body))
+    case Region(body, _) => chez.Builtin("with-region", toChez(body))
```
Comment (b-studios) [inline effekt/shared/src/main/scala/effekt/generator/chez/ChezScheme.scala:135]: Ah I see that we also need the answer type here, yes!

**Review by b-studios — APPROVED** [review]

**Comment by b-studios** [comment]
Thanks, good job!

**Pull request merged**